packages/remix-server-runtime/server.ts

```diff
@@ -328,14 +337,36 @@ async function handleDocumentRequest({
 
   let headerMatches: RouteMatch<ServerRoute>[] = [];
   let routeLoaderResponses: Record<string, Response> = {};
+  let routeLoadersDeferred: Record<
+    string,
+    Record<string, Promise<unknown>>
+  > = {};
   let loaderStatusCodes: number[] = [];
   let routeData: Record<string, unknown> = {};
   for (let index = 0; index < matchesToLoad.length; index++) {
     let match = matchesToLoad[index];
     let result = routeLoaderResults[index];
 
     let error = result.status === "rejected" ? result.reason : undefined;
-    let response = result.status === "fulfilled" ? result.value : undefined;
+    let loaderResponse =
+      result.status === "fulfilled" ? result.value : undefined;
+
+    let response: Response | undefined;
+    if (isDeferredResponse(loaderResponse)) {
+      routeLoadersDeferred[match.route.id] = loaderResponse.deferred;
+      let headers = new Headers(loaderResponse.headers);
+      headers.set("Content-Type", "application/json; charset=utf-8");
+      response = json(
+        loaderResponse.initialData,
+        new Response(null, {
```

Comment:
I don't think you need a `new Response` here. `json` takes a `ResponseInit` as 2nd arg.